spec/unit/indirector/rest_spec.rb

```diff
@@ -353,7 +353,7 @@ def save_request(key, instance, options={})
       it "calls get on the connection" do
         request = find_request('foo bar')
 
-        connection.expects(:get).with("#{url_prefix}/test_model/foo%20bar?environment=production&", anything).returns(mock_response('200', 'response body'))
+        connection.expects(:get).with("#{url_prefix}/test_model/foo+bar?environment=production&", anything).returns(mock_response('200', 'response body'))
```

Comment:
So actually, thinking this through a little more, maybe this is a problem just generally? This test seems to imply that the agent will now send `foo+bar` for something with a space, whereas previously it would send `foo%20bar`, but Puppet Server will decode `foo+bar` with a plus, rather than a space.

In the Rack handler, it seems like we're using `CGI.unescape` on the path - https://github.com/puppetlabs/puppet/pull/5738/files/626913a576da17d4823f084936940e040e5d7b2b#diff-031bcd92260c9c0f6e0440c6710552f5R90, which has the behavior of decoding `+` to space. Since this is only happening in the Rack handler, we might see different behavior in Puppet Server than in the unit tests here.

As far as I can tell, we're not explicitly doing any decoding of the path in Puppet Server itself. This might be automatically happening in Jetty or possibly in our Jetty library - I haven't managed to dig into it yet. In any case, the problem here seems not so much one of backwards compatibility, but rather that it's currently not compatible, at least if we expect to have the same behavior.

With that said though, is there anywhere where we can have a `key` that allows a space? It seems like files might cause a problem there, both in some of the `/file*` endpoints and also in a resource `find` with a file resource with a space in its path.